boards: acrn: acrn: Add support for ADL-n board

akanisetti opened the pull request:
Add support to build zephyr to run on ADL-n board
as a VM on ACRN hypervisor.

Signed-off-by: Anisetti Avinash Krishna <anisetti.avinash.krishna@intel.com>

akanisetti committed:
boards: acrn: acrn: Add support for ADL-n board

Add support to build zephyr to run on ADL-n board
as a VM on ACRN hypervisor.

Signed-off-by: Anisetti Avinash Krishna <anisetti.avinash.krishna@intel.com>

akanisetti commented on boards/acrn/acrn/Kconfig:
```diff
@@ -15,3 +15,10 @@ config BOARD_ACRN_EHL_CRB
 	help
 	  "ACRN User OS on ElkhartLake CRB"
+
+config BOARD_ACRN_ADL_CRB
+	bool
+	select CPU_HAS_FPU
+	select X86_64
+	help
+	  "ACRN User OS on AlderLake CRB"
```
Updated, please review

akanisetti replied:
If I remove `bool` I am getting build error. 
BOARD_ACRN (defined at boards/acrn/acrn/Kconfig:5) defined without a type

nordicjm commented on boards/acrn/acrn/Kconfig:
```diff
@@ -7,11 +7,18 @@ config BOARD_ACRN
-	  "ACRN User OS on ElkhartLake CRB"
+	  ACRN User OS on ElkhartLake CRB
+
+config BOARD_ACRN_ADL_CRB
```
will need to change this file to something like:
```
config BOARD_ACRN_COMMON
	bool
	default y if BOARD_ACRN || BOARD_ACRN_EHL_CRB || BOARD_ACRN_ADL_CRB
	select CPU_HAS_FPU
	select X86_64
```

akanisetti committed:
boards: acrn: acrn: Updated soc name for acrn_ehl_crb

Updated soc name for acrn_ehl_crb to atom as
CONFIG_ATOM is selected for ACRN_EHL_CRB in
Konfig.acrn_ehl_crb but given soc elkhart_lake
in board.yml

Added common config ACRN_COMMON to enable common
configs of ACRN.

Signed-off-by: Anisetti Avinash Krishna <anisetti.avinash.krishna@intel.com>

akanisetti commented on boards/acrn/acrn/board.yml:
```diff
@@ -8,4 +8,9 @@ boards:
   - name: acrn_ehl_crb
     full_name: ACRN on EHL hypervisor
     socs:
-      - name: elkhart_lake
```
While updating for ADL, EHL soc was named elkhart_lake but ATOM soc was selected in Kconfig.acrn_ehl_crb thought this minor change could go through in this PR.

I will create another PR for EHL and common changes and remove from this to focus only on ADL changes.

edersondisouza replied:
Oh, no need for another PR - just a preparation patch in this PR should suffice.
